File: project/backend-services/smart-irrigation-backend/infrastructure/endpoint/graphql/src/main/java/pt/sensae/services/smart/irrigation/backend/infrastructure/endpoint/graphql/model/data/StoveSensorDataHistoryDetails.java
package pt.sensae.services.smart.irrigation.backend.infrastructure.endpoint.graphql.model.data;

public record StoveSensorDataHistoryDetails(String id,
                                            Long reportedAt,
                                            TemperatureDataDetails temperature,
                                            HumidityDataDetails humidity) implements SensorDataHistoryDetails {
}

File: project/frontend-services/libs/smart-irrigation/model/src/lib/model/history/StoveSensorDataHistoryDetails.ts
import {SensorDataHistoryDetails} from "./SensorDataHistoryDetails";
import {TemperatureDataDetails} from "../details/TemperatureDataDetails";
import {HumidityDataDetails} from "../details/HumidityDataDetails";

export class StoveSensorDataHistoryDetails extends SensorDataHistoryDetails {
  constructor(id: string, reportedAt: Date, public temperature: TemperatureDataDetails, public humidity: HumidityDataDetails) {
    super(id, reportedAt)
  }
}
